### ISSUE TITLE
[ browserify / module-deps ] Large UTF8 encoded file can't be compiled correctly

### ISSUE THREAD

**Issue by yhanwen** [substack/node-browserify#1457]
When I try to compile a project which includes a large Angular template file, strange character `�` appeared in the final output. For example, I may get code like:
```
<p>
<span>36氪平台公司仅为您提供在线交流、互联网私募股权投资等网络平台服���，36氪</span>
</p>
```
Then I tracked the code and found the reason for this issue in `module-deps`:

`index.js` line:161:
``` javascript
    var rs = fs.createReadStream(file);
```
When I add option to createReadStream, the problem is resolved. The code is like:
``` javascript
    var rs = fs.createReadStream(file,{
        encoding:'utf8'
    });
```

**Comment by substack** [comment]
Thanks for figuring this out! Do you have the original string that trips up browserify/module-deps? I would like to add this failing case to the test suite.

**Comment by substack** [comment]
I published the suggested fix in module-deps 4.0.4, so you should get this version on a clean browserify install.

**Comment by x36kr** [comment]
This is the file that trips up browserify/module-deps
[templates.js.zip](https://github.com/substack/node-browserify/files/68009/templates.js.zip)
